src/main/java/teammates/ui/webapi/CreateAccountAction.java

```diff
@@ -43,17 +61,23 @@ public JsonResult execute() throws InvalidHttpRequestBodyException {
             log.severe("Unexpected error", e);
             return new JsonResult(e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
         }
+
         List<InstructorAttributes> instructorList = logic.getInstructorsForCourse(courseId);
-        String joinLink = Config.getFrontEndAppUrl(Const.WebPageURIs.JOIN_PAGE)
-                .withRegistrationKey(instructorList.get(0).getKey())
-                .withEntityType(Const.EntityType.INSTRUCTOR)
-                .toAbsoluteString();
-        EmailWrapper email = emailGenerator.generateNewInstructorAccountJoinEmail(
-                instructorList.get(0).getEmail(), instructorName, joinLink);
-        emailSender.sendEmail(email);
-
-        JoinLinkData output = new JoinLinkData(joinLink);
-        return new JsonResult(output);
+
+        try {
+            logic.joinCourseForInstructor(instructorList.get(0).getKey(), userInfo.id);
+        } catch (EntityDoesNotExistException ednee) {
+            throw new EntityNotFoundException(ednee);
+        } catch (EntityAlreadyExistsException eaee) {
+            throw new InvalidOperationException(eaee);
+        } catch (InvalidParametersException ipe) {
+            throw new InvalidHttpRequestBodyException(ipe);
+        }
+
+        // Delete account request as it is no longer needed
+        logic.deleteAccountRequest(instructorEmail, instructorInstitution);
```

Comment:
Maybe, instead of deleting account request, we should mark it as something like "registered at <time>" instead. This serves two purpose:
- We can keep track/archive the list of all account requests we have, registered or not
- We do wonder at times the time lag between user requesting an account and registering his/her account, so this would provide some insights

Deleting account request still has legitimate usages (e.g. for data used in tests or removing wrongly created entities), but the action here should be different.